tests/rptest/tests/acls_test.py

```diff
@@ -321,6 +321,91 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3)
+    @parametrize(authn_method='sasl')
+    @parametrize(authn_method='mtls_identity')
+    def test_security_feature_migration(self, authn_method: str):
+        def check_perms():
+            # run a few times for good health
+            for _ in range(2):
+                try:
+                    self.get_client("base").acl_list()
+                    self.logger.debug(
+                        'list acls should have failed for base user')
+                    return False
+                except ClusterAuthorizationError:
+                    pass
+
+                # should succeed
+                try:
+                    self.get_client("cluster_describe").acl_list()
+                except Exception:
+                    self.logger.debug(
+                        'list acls should have passed for cluster user')
+                    return False
+
+                try:
+                    self.get_super_client().acl_list()
+                except Exception:
+                    self.logger.debug(
+                        'list acls should have passed for super user')
+                    return False
+
+                return True
+
+        self.prepare_cluster(use_tls=True, use_sasl=True)
+
+        wait_until(check_perms,
+                   timeout_sec=90,
+                   err_msg='check perms failed before migration')
+        '''
+        Change the authn method and initiate feature migration
+        with a rolling restart. Do not recreate tls certs.
+        '''
+        self.security.endpoint_authn_method = authn_method
+        self.redpanda.set_security_settings(self.security, init_tls=False)
+        self.logger.debug('Initiating rolling restart')
+        self.redpanda.rolling_restart_nodes(self.redpanda.nodes,
+                                            start_timeout=90,
+                                            stop_timeout=90)
+
+        # Check that the authn_method was set on all brokers
+        pattern = 'Started Kafka API server.*:{' + authn_method + '}.*'
+        assert self.redpanda.search_log(pattern,
+                                        node_matches=len(self.redpanda.nodes))
+        '''
+        Once restart is complete, change cluster wide configs for
+        kafka_enable_authorization and kafka_mtls_principal_mapping_rules
+        via the admin api. Do not expect a restart.
+
+        NOTE:
+        - kafka_enable_authorization overrides sasl in an authorization context
+        - authentication_method overrides sasl in an authentication context
+        '''
```

Comment:
is this a pythonism for multi-line comments (as opposed to `# ...`) because it seems like this technically creates a string and then throws it away.

Reply:
I think multi-line comments should generally be:
```
# line 1
...
# line 100
```

Reply:
> is this a pythonism for multi-line comments (as opposed to `# ...`) because it seems like this technically creates a string and then throws it away.

That is correct.



> I think multi-line comments should generally be:
> 
> ```
> # line 1
> ...
> # line 100
> ```

Sure, I'll change.

Reply:
Done.